src/main/java/com/flightstats/hub/metrics/StatsDFilter.java

```diff
@@ -50,15 +50,16 @@ public boolean isTestChannel(String channel) {
     }
 
     boolean isSecondaryReporting(String name) {
-        Optional<Webhook> optionalWebhook = Optional
-                .ofNullable(webhookConfigDao.getCached(name))
-                .filter(Webhook::isSecondaryMetricsReporting);
+        if (name.equals("request")) return false;
 
-        Optional<ChannelConfig> optionalChannelConfig = Optional
+        return Optional
```

Comment:
I think I would have left this as two variables instead of inlining them... I know the fact that you can't duck-type these is annoying. Another option might be to do something like...  (note: almost certainly has typos:)
```
Optional<Boolean> webhookHasSecondaryReporting = Optional.ofNullable(webhookConfigDao.getCached(name))
    .map(Webhook::isSecondaryMetricsReporting)
    .orElse(false);
boolean channelHasSecondaryReporting = Optional.ofNullable(channelConfigDao.getCached(name))
    .map(ChannelConfig::isSecondaryMetricsReporting)
    .orElse(false);
return webhookHasSecondaryReporting || channelHasSecondaryReporting;
```
Assuming the DAOs don't have to follow a generic interface, I would also have probably extracted the two variables out as methods into their respective DAOs so they could be tested slightly more directly.

Reply:
There is a specific design reason that I rewrote it this way:
  In the previous fashion, being that it is a black box whether the string we are comparing is a channel name or a webhook name or (very rarely now) neither both the the channel dynamo table and the webhook dynamo table could be queried before resolving the boolean. By doing it inline and putting the webhook evaluation second, if the channel matches, the extra call to the webhook dao (and the webhook cache and possibly the webhook dynamo table, which is set at half the throughput of the channel table) will not happen, saving a decent amount of overhead. Otherwise, I 100% agree that it was more readable and clean before.

All DAO classes implement the generic Dao interface :(

Reply:
The short-circuiting makes sense! One way to do that rather than with an or operator (which was the surprising/unintuitive part of that statement, I think) would be to drop the first `.orElse(false)` and do an `.orElse(splat-all-the-code-from-channel-checking-here.orElse(false))`. That way all the behavior is Java 8 Optional stuff, but it's nested and it's still kinda bleh.

That said, I think your idea of extracting the methods to get a balance of readability and performance is great. 👍

Reply:
Maybe not relevant here, but there is Optional.orElseGet(Supplier) which can be nice if you need a short circuit. If you do something like Optional.of(thing).orElse(someMethod()), the someMethod() always gets called which is not necessarily intuitive. Well, it's obvious from a compiler viewpoint, but not from a human reading it viewpoint. I've gotten bitten by that a few times.

Sorry if that's old news or completely irrelevant.

Reply:
I forgot `.orElse` didn't short-circuit. Thanks!